src/Nethermind/Nethermind.Serialization.Rlp/HeaderDecoder.cs

```diff
@@ -216,9 +211,9 @@ public void Encode(RlpStream rlpStream, BlockHeader? header, RlpBehaviors rlpBeh
                 }
             }
 
-            if (header.Number >= Eip1559TransitionBlock)
+            if (header.BaseFeePerGas is not null)
             {
-                rlpStream.Encode(header.BaseFeePerGas);
+                rlpStream.Encode(header.BaseFeePerGas.Value);
             }
 
             if (header.WithdrawalsRoot is not null || header.ExcessDataGas is not null)
```

Comment:
Here we check for null to encode, so I believe with basefee we should work the same way